src/components/SlideButton/SlideButton.js

```diff
@@ -0,0 +1,142 @@
+import { ChevronsRight } from '@lana/b2c-mapp-ui-assets';
+
+const components = {
+  ChevronsRight,
+};
+
+const props = {
+  initialInstructionText: {
+    type: String,
+    default: 'Desliza para confirmar',
+  },
+  completedText: {
+    type: String,
+    default: 'Confirmado',
+  },
+};
+
+const methods = {
+  startSlide(event) {
+    if (this.completed) { return; }
+    this.initialMouseX = this.getMouseXPosFromEvent(event);
+    this.endPoint = this.getEndingPoint();
+    this.calculateSliderInitialWidth();
+    this.calculateSlideButtonInitialPosition();
+    this.updateSlideButton(0);
+    this.updateSlider(0);
+    this.startDrag = true;
+    this.sliderClass = 'started';
+  },
+  getEndingPoint() {
+    const clientRects = this.$refs.slider.getClientRects()[0];
+    return clientRects.right;
+  },
+  calculateSliderInitialWidth() {
+    const sliderLeftPos = this.$refs.slider.getClientRects()[0].x;
+    this.initialSliderWidth = this.initialMouseX - sliderLeftPos;
+    if (this.initialSliderWidth < 0) {
+      this.initialSliderWidth = 0;
+    }
+  },
+  calculateSlideButtonInitialPosition() {
+    this.initialSlideButtonPosition = this.$refs.slider.getClientRects()[0].x;
+  },
+  continueSlide(event) {
+    if (!this.startDrag) { return; }
+    this.currentMouseX = this.getMouseXPosFromEvent(event);
+    const delta = this.currentMouseX - this.initialMouseX;
+    this.updateSlider(delta);
+    this.updateSlideButton(delta);
+    if (this.sliderReachedEndPoint()) { this.endSlide(); }
+  },
+  endSlide() {
+    this.startDrag = false;
+    if (this.sliderReachedEndPoint()) {
+      this.sliderClass = 'completed';
+      this.overlayStyle.width = `${this.$refs.slider.getClientRects()[0].width}px`;
+      this.actionConfirmed();
+    } else {
```

Comment:
(no-else) please add a new `return;` statement after line: 57, and then remove this unnecessary `else` wrapper